### PR TITLE
Adding storage information

### PR DESCRIPTION
**Pull request by ancorgs**
Most of the user-facing documentation of the storage area was directly at the Agama source code repository and interleaved with development notes and documentation of the design decisions.

This pull request does two things:

- Moving all the user-facing documentation to this repository.
- Updating the web UI documentation to reflect what we are currently developing, instead of the current interface which is in the process of being replaced.

See the [preview](https://agama-preview-pull-32.surge.sh/) for easier review.

This goes together with https://github.com/agama-project/agama/pull/1841, which is the counterpart at the Agama repository.

### CODE REVIEW

File: docs/user/unattended/storage.md [hunk 1]
```diff
@@ -0,0 +1,956 @@
...
+  cipher [<string>]
+  keySize [<number>]
+```
+
+To illustrate how all that fits together, let's see the following example in which the first disk of
+the system is partitioned and a volume group is created on top of that partition (after encrypting
+it) to allocate two file systems.
+
+```json
+"storage": {
```
Comment (imobachgs) [inline docs/user/unattended/storage.md:199]: Please, let's use the same indentation in all the examples (2 spaces).

File: docs/user/unattended/storage.md [hunk 2]
```diff
@@ -0,0 +1,956 @@
...
+
+```
+Storage
+  drives <Drive[]>
+  volumeGroups <VolumeGroup[]>
+  mdRaids <MdRaid[]>
+  btrfsRaids <BtrfsRaid[]>
+  nfsMounts <NFS[]>
+  boot [BootSettings]
+```
```
Comment (joseivanlopez) [inline docs/user/unattended/storage.md:49]: I would avoid this "formal" specification in the users' documentation. Instead, I would use incomplete examples with references to the schema:

~~~
"storage": {
  "drives": [ ... ],
  "volumeGroups": [ ... ],
  "mdRaids": [ ... ]
}
~~~

File: docs/user/unattended/storage.md [hunk 3]
```diff
@@ -0,0 +1,956 @@
...
+
+```json
+{ "search": "*" }
+
+{ "search": { "ifNotFound": "skip" } }
+```
+
+## Referencing Other Devices
+
+Sometimes is necessary to reference other devices as part of the specification of an LVM volume
```
Comment (imobachgs) [inline docs/user/unattended/storage.md:398]:
```suggestion
Sometimes it is necessary to reference other devices as part of the specification of an LVM volume
```

File: docs/user/unattended/storage.md [hunk 4]
```diff
@@ -0,0 +1,956 @@
...
+            "logicalVolumes": [
+                { "filesystem": { "path": "/" } },
+                { "filesystem": { "path": "swap" } }
+            ]
+        }
+    ]
+}
+```
+
+Assuming the product configuration specifies a root filesystem with a minimum size of 5 GiB and a
```
Comment (imobachgs) [inline docs/user/unattended/storage.md:539]: What "product configuration" means? OK, I know what it means... but what about a user that knows nothing about Agama? Adding a footnote should be enough.

File: docs/user/unattended/storage.md [hunk 5]
```diff
@@ -0,0 +1,956 @@
...
+max of 20 GiB and sets that the swap must have a size equivalent to the RAM on the system, then
+those values would be applied to the logical volumes and the partition with alias "pv" would be
+sized accordingly, taking into account all the overheads and roundings introduced by the different
+technologies like LVM or the used partition table type.
+
+## Partitions Needed for Booting
+
+Using a `boot` entry makes it possible to configure whether (and where, using an alias) Agama
+should calculate and create the extra partitions needed for booting. If the device is not
+specified, Agama will take the location of the root file system as a reference.
```
Comment (imobachgs) [inline docs/user/unattended/storage.md:549]: What about adding an example?

File: docs/user/unattended/storage.md [hunk 6]
```diff
@@ -0,0 +1,956 @@
...
+
+## Keeping an Existing File System or Encryption Layer
+
+The entries for both `encryption` and `filesystem` contain a flag `reuse` with a default value of
+false. It can be used in combination with `search` to specify the device must not be re-encrypted
+or re-formatted.
```
Comment (imobachgs) [inline docs/user/unattended/storage.md:555]: Please, add an example.

File: docs/user/unattended/storage.md [hunk 7]
```diff
@@ -0,0 +1,956 @@
...
+"storage": {
+  "drives": [
+    { "search": "/dev/sda", "alias": "sda" },
+    { "search": "/dev/sdb", "alias": "sdb" },
+  ],
+  "mdRaids": [
+    {
+      "devices": [
+        {
+          "generate": {
```
Comment (imobachgs) [inline docs/user/unattended/storage.md:734]: I am not sure whether the concept of generation ("generate") deserves some introduction.

**Review by imobachgs — COMMENTED** [review]
The documentation looks great, but I have the impression that it might be too much for a user. Perhaps this document is just fine and we just need another one which is oriented to use cases.

File: docs/user/unattended/storage.md (hunk 1 above)
Comment (ancorgs) [reply docs/user/unattended/storage.md:104]: Fixed.

File: docs/user/unattended/storage.md (hunk 2 above)
Comment (ancorgs) [reply docs/user/unattended/storage.md:50]: Done.

File: docs/user/unattended/storage.md [hunk 8]
```diff
@@ -0,0 +1,956 @@
...
+                    "size":   "2 GiB",
+                    "filesystem": { "path": "swap", "type": "swap" }
+                }
+            ]
+        }
+    ]
+}
+```
+
+## Searching Existing Devices
```
[the remark this answers is not in the document]
Comment (ancorgs) [reply docs/user/unattended/storage.md:246]: As discussed offline. Too much for this pull request.

I removed the formal specification and restructured a bit the information to start with a couple of simple examples. But many of the other suggested improvements will have to wait for subsequent pull requests.

File: docs/user/unattended/storage.md (hunk 3 above)
Comment (ancorgs) [reply docs/user/unattended/storage.md]: Changed.

File: docs/user/unattended/storage.md (hunk 5 above)
Comment (ancorgs) [reply docs/user/unattended/storage.md:537]: Added.

File: docs/user/unattended/storage.md (hunk 6 above)
Comment (ancorgs) [reply docs/user/unattended/storage.md:564]: In the future. ;-)

File: docs/user/unattended/storage.md (hunk 7 above)
Comment (ancorgs) [reply docs/user/unattended/storage.md:743]: Maybe... for a future pull request. :-)

File: docs/user/unattended/storage.md (hunk 4 above)
Comment (ancorgs) [reply docs/user/unattended/storage.md:527]: I added a sentence to clarify that in a previous paragraph.

**Pull request merged**